pkg/process/net/common.go

```diff
@@ -59,21 +58,15 @@ type RemoteSysProbeUtil struct {
 	httpClient http.Client
 }
 
-// SetSystemProbePath sets where the System probe is listening for connections
-// This needs to be called before GetRemoteSystemProbeUtil.
-func SetSystemProbePath(path string) {
-	globalSocketPath = path
-}
-
 // GetRemoteSystemProbeUtil returns a ready to use RemoteSysProbeUtil. It is backed by a shared singleton.
-func GetRemoteSystemProbeUtil() (*RemoteSysProbeUtil, error) {
-	err := CheckPath()
+func GetRemoteSystemProbeUtil(path string) (*RemoteSysProbeUtil, error) {
```

Comment:
Nit: One thing you could do to hide complexity a bit here is instead pass in the config object. Like
```go
func GetRemoteSystemProbeUtil(syscfg config.ConfigReader) (*RemoteSysProbeUtil, error) {
   path := syscfg.GetString("system_probe_config.sysprobe_socket")
   ...
}
```
This makes things a bit more mockable and IMO encapsulates the actual config logic a bit better so the caller doesn't have to worry about what config setting they need to use.

Reply:
Unfortunately that would be quite a large change, and after discussion we don't think it's worth tackling now.

The `ProcessCheck` doesn't have access to a config that implements `ConfigReader`, as `Check.Init()` takes a [`*SysProbeConfig`](https://github.com/DataDog/datadog-agent/blob/85313607b6ec7a846e78b396833248f8070cfa2f/pkg/process/checks/checks.go#L29-L36) object instead of the full config object.

The `ConnectionsCheck`, in addition to the above `*SysProbeConfig` passed in via `Init()`, has access to [its own system-probe config object](https://github.com/DataDog/datadog-agent/blob/85313607b6ec7a846e78b396833248f8070cfa2f/cmd/system-probe/config/config.go#L52-L72), which also does not implement `ConfigReader`.